include/dlaf/eigensolver/tridiag_solver/rot.h

```diff
@@ -0,0 +1,243 @@
+    // keep alive the entire chunk, so also the part of other tiles, through its memory view).
+    // Anyway, it is more clean to get them all, and then just here use a single one to access the
+    // full column.
+    matrix::Tile<T, D>& tile_ws = all_ws[0];
+
+    // Note:
+    // The entire algorithm relies on a strong assumption about memory layout of all tiles involved,
+    // i.e. both the input tiles selected from the matrix and the tiles for the workspace.
+    // By relying on the fact that all of them are stored in a column-layout, we can work on column
+    // vectors ignoring their distribution over different tiles, and it would be enough just getting
+    // the pointer to the top head of the column and all other data can be easily accessed since it
+    // is stored contiguously after that.
+    // Ignoring tile organization of the memory comes handy also for communication. Indeed, during
+    // column exchange, just a single MPI operation per column is issued, instead of communicating
+    // independently the parts of column from each tile.
+    auto getColPtr = [&dist_sub, &tiles, &tile_ws](const SizeType col_index, const bool hasIt) -> T* {
+      if (hasIt) {
+        const LocalTileIndex tile_col{dist_sub.nextLocalTileFromGlobalElement<Coord::Row>(0),
+                                      dist_sub.nextLocalTileFromGlobalElement<Coord::Col>(col_index)};
+        const std::size_t linear_tile_col = to_sizet(dist_sub.localTileLinearIndex(tile_col));
+        return tiles[linear_tile_col].ptr(dist_sub.tileElementIndex({0, col_index}));
+      }
+      return tile_ws.ptr({0, 0});
+    };
+
+    // Change SoR for the rotations to just the range, by removing the offset
+    std::transform(rots.begin(), rots.end(), rots.begin(),
+                   [offset = i_begin * mb](const GivensRotation<T>& rot) {
+                     return GivensRotation<T>{rot.i - offset, rot.j - offset, rot.c, rot.s};
+                   });
+
+    const SizeType m = dist_sub.localSize().rows();
+
+    for (const GivensRotation<T>& rot : rots) {
+      const SizeType j_x = rot.i / mb;
+      const SizeType j_y = rot.j / mb;
+
+      const comm::IndexT_MPI rankColX = dist_sub.template rankGlobalTile<Coord::Col>(j_x);
+      const comm::IndexT_MPI rankColY = dist_sub.template rankGlobalTile<Coord::Col>(j_y);
+
+      const bool hasX = dist_sub.rankIndex().col() == rankColX;
+      const bool hasY = dist_sub.rankIndex().col() == rankColY;
+
+      if (!hasX && !hasY)
+        continue;
+
+      T* col_x = getColPtr(rot.i, hasX);
+      T* col_y = getColPtr(rot.j, hasY);
+
+      const bool hasBothXY = hasX && hasY;
+
+      std::vector<ex::unique_any_sender<>> cps;
+      if (!hasBothXY) {
+        // TODO compute TAG
+
+        const comm::IndexT_MPI rank_partner = hasX ? rankColY : rankColX;
+
+        // TODO possible optimization, check if it is zero or not
```

Comment:
I would just go for dropping this TODO without creating an issue for tracking this chance. This algorithm implementation is a very basic one, and we might evaluate in the future if it is worth improving it, and in case how (there are many things that can be done in a smarter way).

Reply:
Yes, I think the TODO can be dropped.